include/tvm/target/target.h

```diff
@@ -44,6 +44,8 @@ class TargetNode : public Object {
  public:
   /*! \brief The kind of the target device */
   TargetKind kind;
+  /*! \brief Target host information of the target device */
+  Optional<ObjectRef> host;
```

Comment:
Should this be `Optional<Target>`?

Reply:
It is not doable, because Target is not defined yet 😆

Reply:
Ah...good point. I checked the constructor and it does check the host type to be `Target`, so we're good here. A nit might be improving the comment of `host` to explicitly say it must be a `Target`.

Reply:
Sure. Fixed.

Reply:
could we reverse the order and forward-declare TargetNode?